www/nginx/src/opnsense/mvc/app/views/OPNsense/Nginx/index.volt

```diff
@@ -223,16 +223,17 @@
         <table id="grid-location" class="table table-condensed table-hover table-striped table-responsive" data-editDialog="locationdlg">
             <thead>
             <tr>
-                <th data-column-id="description" data-type="string" data-sortable="true"  data-visible="true">{{ lang._('Description') }}</th>
-                <th data-column-id="urlpattern" data-type="string" data-sortable="true"  data-visible="true">{{ lang._('URL Pattern') }}</th>
-                <th data-column-id="path_prefix" data-type="string" data-sortable="true"  data-visible="true">{{ lang._('URL Path Prefix') }}</th>
-                <th data-column-id="matchtype" data-type="string" data-sortable="true"  data-visible="true">{{ lang._('Match Type') }}</th>
-                <th data-column-id="upstream" data-type="string" data-sortable="true"  data-visible="false">{{ lang._('Upstream') }}</th>
-                <th data-column-id="waf_status" data-type="string" data-sortable="true"  data-visible="true">{{ lang._('WAF Status') }}</th>
-                <th data-column-id="xss_block_score" data-type="string" data-sortable="true"  data-visible="false">{{ lang._('XSS Score') }}</th>
-                <th data-column-id="sqli_block_score" data-type="string" data-sortable="true"  data-visible="false">{{ lang._('SQLi Score') }}</th>
-                <th data-column-id="custom_policy" data-type="string" data-width="50%" data-sortable="true"  data-visible="false">{{ lang._('WAF Policies') }}</th>
-                <th data-column-id="force_https" data-type="boolean" data-width="10em" data-sortable="true"  data-visible="true">{{ lang._('Force HTTPS') }}</th>
+                <th data-column-id="uuid" data-type="string" data-sortable="true" data-visible="false">{{ lang._('ID') }}</th>
```

Comment:
I don't think it makes much sense to display this column in the overview. Nobody knows the meaning of the uuid.
I know that it may make sense to know it when you add a plugin config via CLI, but then it would make more sense to add a copy uuid to clipboard button to the commands if possible.
The buttons are added here: https://github.com/opnsense/plugins/blob/9526844aad5c66d0659629387a8a68e69ede6b6f/www/nginx/src/opnsense/www/js/nginx/src/nginx_config.js#L86

you will need to rebuild the javascript.

@fichtner what do you think?

Reply:
thanks for review
actually, I already use this column for quick folder with location\server inclusion(s) matching (and for copying uuid in the browser for using it to create a dir in shell). I agree that not everyone understands the purpose of the column, so it is not displayed by default (but at the same time, there are enough questions on the forum about configurations that require the use of hooks)

adding the ability to copy the ID seems like a good idea to me, but maybe it can be done in a several steps? adding uuid-copy button to opnsense_bootgrid_plugin.js can take a while imho

Reply:
another rationale for the usefulness of a uuid column: in 1.32 `$internalModelUseSafeDelete` is enabled and for example, deleting an ACL used in the http server may result in a message like "Item in use. Nginx - {Nginx.http_server.728012f5-11ee-4fe3-ac61-7d8fb92420ad}". So ID column can\will be convenient

Reply:
since it is only one line of javascript and one line of html, I think it is pretty easy doable:
* One line to add in the commands - see link above
* https://www.w3.org/TR/clipboard-apis/#dom-clipboard-writetext

Reply:
@fabianfrz you mean like https://github.com/kulikov-a/plugins/commit/da51766dd0767ff235de8a35d8080199251fc0d5#diff-bedd1baa59f13b150070b4917a4cb5a67f1a10beb5bb8b06dd345bad4099fe21 ?
yes, it works.
I don't know for sure if any other plugin uses uuid-hooks, and what @fichtner  will say to that, but perhaps in this case it is better to place the uuid-copy command in the `opnsense_bootgrid_plugin` right away?
but please, let's still allow the display of the uuid-column (and add the `uuid` to the api search base)!
I tried it and it turned out to be so handy (especially together with gui-config display) that I even added it to the production servers ;)

Reply:
Can’t columns be hidden by default?

Reply:
it is ) (`data-visible="false"`)

Reply:
Ok they are. Biggest issue here is doing two things here in the PR and the less descriptive part of it changes a number of lines. Perhaps splitting would have made the intent clearer, but likely the same discussion would start with the exception of the socket PR being merged already 😉

Reply:
yep, I'm also starting to think that it makes sense to limit this pr to the setup script and the Cache Path grid columns naming. moving the rest to the 1.33 pr ).
can we continue the uuid-discussion here so we don't have to start all over again later?

Reply:
Then split out socket change to new PR and rename this one to keep UUID discussion?

Reply:
@kulikov-a Yes, I meant exactly this. So that is most likely the most handy version of it.

Reply:
Ok, I wouldn’t place the copy button next to the delete button ;)

Reply:
(Don’t think there is need for a generic uuid copy button in core)

Reply:
@fabianfrz @fichtner thanks! buttons commit added to pr: "Copy UUID to clipboard" added between Edit and Clone buttons, buttons styles aligned with core buttons style, Command column width adjusted, 'uuid' added to the search bases of the relevant api actions